shell/src/main/java/alluxio/cli/fs/command/DuCommand.java

```diff
@@ -47,45 +88,117 @@ public String getCommandName() {
   @Override
   protected void runPlainPath(AlluxioURI path, CommandLine cl)
       throws AlluxioException, IOException {
-    long sizeInBytes = getFileOrFolderSize(mFileSystem, path);
-    System.out.println(path + " is " + sizeInBytes + " bytes");
+
+    ListStatusOptions listOptions = ListStatusOptions.defaults().setRecursive(true);
+    List<URIStatus> statuses = mFileSystem.listStatus(path, listOptions);
+    if (statuses == null || statuses.size() == 0) {
+      return;
+    }
+
+    if (cl.hasOption(HEADER_OPTION_NAME)) {
+      printInfo("File Size", "In Alluxio",
+          cl.hasOption(MEMORY_OPTION_NAME) ? "In Memory" : "", "Path");
+    }
+
+    getSizeInfo(path, statuses, cl.hasOption(READABLE_OPTION_NAME),
+        cl.hasOption(SUMMARIZE_OPTION_NAME), cl.hasOption(MEMORY_OPTION_NAME));
   }
 
   /**
-   * Calculates the size of a path (file or folder) specified by a {@link AlluxioURI}.
+   * Gets and prints the size information of the input path according to options.
    *
-   * @param fs a {@link FileSystem}
-   * @param path a {@link AlluxioURI} denoting the path
-   * @return total size of the specified path in byte
+   * @param path the path to get size info of
+   * @param statuses the statuses of files and folders
+   * @param readable whether to print info of human readable format
+   * @param summarize whether to display the aggregate summary lengths
+   * @param addMemory whether to display the memory size and percentage information
    */
-  private long getFileOrFolderSize(FileSystem fs, AlluxioURI path)
-      throws AlluxioException, IOException {
-    long sizeInBytes = 0;
-    ListStatusOptions listOptions = ListStatusOptions.defaults().setRecursive(true);
-    List<URIStatus> statuses = fs.listStatus(path, listOptions);
-    for (URIStatus status : statuses) {
-      if (!status.isFolder()) {
-        sizeInBytes += status.getLength();
+  private void getSizeInfo(AlluxioURI path, List<URIStatus> statuses,
+      boolean readable, boolean summarize, boolean addMemory) {
+    if (summarize) {
+      long sizeInBytes = 0;
+      long sizeInAlluxio = 0;
+      long sizeInMem = 0;
+      for (URIStatus status : statuses) {
+        if (!status.isFolder()) {
+          long size = status.getLength();
+          sizeInBytes += size;
+          sizeInMem += size * status.getInMemoryPercentage() / 100;
```

Comment:
the many integer divisions could cause issues with rounding if there are lots of small files. Can we wait until the end to do the division?

Reply:
Remove to later, so we only need to divide for once if -s is given. Thanks!